csrc/expr_simplifier.cpp

```diff
@@ -188,6 +189,26 @@ class Context {
         assumptions.push_back(bop->rhs());
       }
     }
+
+    // Check for obvious contradictions. Could have helped debugging #1572.
+    // Let me know if there's a subquadratic way to check this. `Val*` equality
+    // was't sufficient, so I had to call `sameAs` instead.
+    for (const auto& [a, b] : getKnownLessThan()) {
+      for (const auto& [x, y] : getKnownLessEqual()) {
+        // a < b && b <= a is impossible.
```

Comment:
We could add an inner loop to check `a < b && b < a`.

Reply:
Done.